### ISSUE TITLE
Provide indexing status report

### ISSUE THREAD

**Issue by gothub** [NCEAS/metacat#1555]
As indexing occurs after `MNStorage.create()` or `MNStorage.update()` calls have returned status to the client, the client has no way to determine the cause of an indexing failure if one occurs.

Should we consider having metacat retain indexing status and error messages, that could then be retrieved by the user via a new DataONE MN call?

I'll provide details of a possible implementation if we decide that this should be done.

BTW - I didn't see any previous issues on this topic, but they may have been missed.

**Comment by mbjones** [comment]
👍 I think its a good idea. In general, we don't have good facilities for providers to understand what processing has happened, whether there were errors, and whether the provider can fix these errors. We proposed a new event notification service for this type of information for the Arctic Data Center, so this type of API would clearly fit within that type of work, but which would likely go beyond just indexing notifications and also include other notifications, such as version changes on datasets for users.

**Comment by laurenwalker** [comment]
I also agree that this would be a great service that would fix client issues such as: nceas/metacatui#569, nceas/metacatui#610, nceas/metacatui#558, nceas/metacatui#339, nceas/metacatui#1425
